src/NuGet.Core/NuGet.ProjectModel/Utf8JsonReaderExtensions.cs

```diff
@@ -0,0 +1,231 @@
+        //    bool wasRead;
+
+        //    while ((wasRead = reader.Read()) && reader.TokenType == JsonTokenType.Comment)
+        //    {
+        //    }
+
+        //    return wasRead;
+        //}
+
+        //internal static bool ReadNextTokenAsBoolOrFalse(this ref Utf8JsonReader reader)
+        //{
+        //    if (reader.ReadNextToken() && (reader.TokenType == JsonTokenType.False || reader.TokenType == JsonTokenType.True))
+        //    {
+        //        return reader.GetBoolean();
+        //    }
+        //    return false;
+        //}
+
+        //internal static string ReadNextTokenAsString(this ref Utf8JsonReader reader)
+        //{
+        //    if (ReadNextToken(ref reader))
+        //    {
+        //        return ReadTokenAsString(ref reader);
+        //    }
+
+        //    return null;
+        //}
+
+        //internal static IList<T> ReadObjectAsList<T>(this ref Utf8JsonReader reader, JsonSerializerOptions options)
+        //{
+        //    if (reader.TokenType != JsonTokenType.StartObject)
+        //    {
+        //        return new List<T>(0);
+
+        //    }
+        //    //We use JsonObjects for the arrays so we advance to the first property in the object which is the name/ver of the first library
+        //    reader.ReadNextToken();
+
+        //    if (reader.TokenType == JsonTokenType.EndObject)
+        //    {
+        //        return new List<T>(0);
+        //    }
+
+        //    var objectConverter = (JsonConverter<T>)options.GetConverter(typeof(T));
+        //    var listObjects = new List<T>();
+        //    do
+        //    {
+        //        listObjects.Add(objectConverter.Read(ref reader, typeof(T), options));
+        //        //At this point we're looking at the EndObject token for the object, need to advance.
+        //        reader.ReadNextToken();
+        //    }
+        //    while (reader.TokenType != JsonTokenType.EndObject);
+        //    return listObjects;
+        //}
+
+        //internal static List<string> ReadNextStringArrayAsList(this ref Utf8JsonReader reader, List<string> strings = null)
+        //{
+        //    ReadNextToken(ref reader);
+        //    return ReadStringArrayAsList(ref reader, strings);
+        //}
+
+        //internal static List<string> ReadStringArrayAsList(this ref Utf8JsonReader reader, List<string> strings = null)
+        //{
+        //    if (reader.TokenType == JsonTokenType.StartArray)
+        //    {
+        //        while (ReadNextToken(ref reader) && reader.TokenType != JsonTokenType.EndArray)
+        //        {
+        //            string value = ReadTokenAsString(ref reader);
+
+        //            strings = strings ?? new List<string>();
+
+        //            strings.Add(value);
+        //        }
+        //    }
+
+        //    return strings;
+        //}
+
+
+        //internal static IList<string> ReadStringArrayAsIList(this ref Utf8JsonReader reader, IList<string> strings = null)
```

Comment:
https://github.com/NuGet/NuGet.Client/pull/5529/files#r1414980091

Reply:
> Can we handle empty and return Array.Empty? Lots of the callers of this pass in a new empty List.
@davkean 
This is currently commented out but I want to address this. The reason I added the option was because some of the classes we're deserializing create an empty list on instantiation, so I was trying to avoid creating multiple empty lists.